### PR TITLE
Add commit analyzer

### PR DESCRIPTION
**Pull request by thenickcox**
This commit adds the commit analyzer to check the JSON front-matter, markdown parsing, and relative links to other files (images and other markdown files) to be run on Travis CI.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,21 @@ so like this (folder names are case-sensitive):
 [API v2.0 Overview](../Getting Started/api-v2-overview.md)
 [Login](../Authentication/login.md )
 ```
+
+### Commit Analyzer
+
+This repository contains a [commit analyzer](https://github.com/thenickcox/commit_analyzer) that runs against each file in the repository validating that the following are true:
```
Comment (nathanyoung) [inline README.md]: Should the commit_analyzer be hosted here? https://github.com/CenturyLinkCloud/KB-Commit-Analyzer

Comment (mattfordham) [inline README.md]: @nathanyoung could you add @thenickcox to the team? We thought you did this, but seems it didn't go through somehow. Thanks!

Comment (thenickcox) [inline README.md]: Yep. I'll transfer ownership.

Comment (nathanyoung) [inline README.md]: invitation sent

Comment (thenickcox) [inline README.md]: Just got an email from GitHub:

```
Hi Nick Cox,

Your invitation to the CenturyLink Cloud organization on GitHub has been canceled.
Links to the invitation in previous emails will no longer work.

Thanks,
The GitHub Team
```
Can you try again?

Comment (nathanyoung) [inline README.md]: done

**Comment by thenickcox** [comment]
Updated the README accordingly.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "lib"]
+	path = lib
+	url = git@github.com:thenickcox/commit_analyzer.git
```
Comment (nathanyoung) [inline .gitmodules]: does this need to change as well?

**Comment by thenickcox** [comment]
@nathanyoung Just wanted to let you know that the idea of this PR (and the corresponding PRs to the other two repos) is to give you the opportunity to give feedback. You can test it locally by:
1) Setting up an account on [travis-ci.org](http://travis-ci.org)
2) [Check out this PR](https://help.github.com/articles/checking-out-pull-requests-locally/)
3) Push to your account on travis
4) View the script output

Please let me know if you'd like anything about the script or output to change. Better yet, file an issue on the [Commit Analyzer repo](https://github.com/CenturyLinkCloud/KB-Commit-Analyzer/issues)!

**Comment by nathanyoung** [comment]
@thenickcox so we just tried this and we're not quite sure what to expect. We made a commit to the APIDocs to update the readme (7fbb3e6093135331d7f5c74add70693b10255f86). Travis started building but then errored.

https://travis-ci.org/CenturyLinkCloud/APIDocs/builds/72164389

**Comment by thenickcox** [comment]
I've seen this happen if git thinks there are any changes to the submodule directory when you push. What happens if you run `git status` from the `lib` directory?

**Comment by nathanyoung** [comment]
Definitely thinks something has changed.

screen shot 2015-07-22 at 11 29 33 am

screen shot 2015-07-22 at 11 30 00 am

**Comment by thenickcox** [comment]
Great. `cd` into lib and run `git reset --hard`. Then try to push master again.

**Comment by nathanyoung** [comment]
Ok, I was able to successfully push, but travis build errored: https://travis-ci.org/CenturyLinkCloud/APIDocs/builds/72170195

screen shot 2015-07-22 at 11 37 27 am

**Comment by thenickcox** [comment]
Okay. That's because the submodule is a private repo. Can you follow the steps [here](http://stackoverflow.com/a/24600210/931934) and let me know the output?

**Comment by nathanyoung** [comment]
There is no way to create a token that is specific to just one repository, or at least not that I know of. So there would be a checked in token that would have access to all private repos not something we want. 

Is there any reason to not just make the comment analyzer a public repo? Do we have secretes or confidential information checked in there?

**Comment by thenickcox** [comment]
The commit analyzer could be public if you want. There is no confidential information. It's just a few scripts. It would just be a matter of if you want it to be open source or not.

**Comment by nathanyoung** [comment]
Alright, so I made the commit analyzer public and re-ran the build. There are 3 errors in the travis log, but the build passed. Theoretically, shouldn't it fail?

https://travis-ci.org/CenturyLinkCloud/APIDocs/builds/72170195

**Comment by thenickcox** [comment]
Let me double-check that it fails for me locally.

**Comment by thenickcox** [comment]
I just reran this and it looks like it pulled in a commit from you that fixes the docs. Were you able to get it to fail?

**Comment by thenickcox** [comment]
Here's what happened when I ran this locally:
screen shot 2015-07-22 at 1 33 10 pm

**Comment by nathanyoung** [comment]
Yeah, I just fixed those. Now we were going to test a PR that breaks something to see if it Travis reports a build failure.

**Comment by nathanyoung** [comment]
So it found the bad link I committed, but the build still passed. 

https://travis-ci.org/CenturyLinkCloud/APIDocs/builds/72189183

**Comment by thenickcox** [comment]
Wondering if the submodule was updated properly. [Here] is a link to my "bad link" commit, which failed as expected.

**Comment by thenickcox** [comment]
@nathanyoung Should we move this to email to make sure none of this conversation slips through the cracks?

**Pull request merged**